appsec/src/helper/remote_config/listeners/asm_features_listener.cpp

```diff
@@ -7,29 +7,38 @@
 #include "../../json_helper.hpp"
 #include "../../utils.hpp"
 #include "../exception.hpp"
-#include <algorithm>
+#include "config_aggregators/asm_features_aggregator.hpp"
 #include <rapidjson/document.h>
 
-void dds::remote_config::asm_features_listener::on_update(const config &config)
+namespace dds::remote_config {
+
+void asm_features_listener::init()
 {
-    rapidjson::Document serialized_doc;
+    ruleset_ = rapidjson::Document(rapidjson::kObjectType);
+    aggregator_ = std::make_unique<asm_features_aggregator>();
+    aggregator_->init(&ruleset_.GetAllocator());
+}
 
-    {
-        const mapped_memory contents{config.read()};
-        if (!json_helper::parse_json(contents, serialized_doc)) {
-            throw error_applying_config("Invalid config contents");
-        }
-    }
+void asm_features_listener::on_update(const config &config)
+{
+    aggregator_->add(config);
+}
+
+void asm_features_listener::commit()
+{
+    aggregator_->aggregate(ruleset_);
 
-    auto asm_itr = json_helper::get_field_of_type(
-        serialized_doc, "asm", rapidjson::kObjectType);
+    auto asm_itr =
+        json_helper::get_field_of_type(ruleset_, "asm", rapidjson::kObjectType);
     if (!asm_itr) {
+        service_config_->unset_asm();
```

Comment:
Why throw here? I don't think the config is mandatory.

Reply:
fixed it